addons/mail/models/mail_message.py

```diff
@@ -176,6 +177,13 @@ def default_get(self, fields):
     mail_ids = fields.One2many('mail.mail', 'mail_message_id', string='Mails', groups="base.group_system")
     canned_response_ids = fields.One2many('mail.shortcode', 'message_ids', string="Canned Responses", store=False)
 
+    def _compute_is_current_user_or_guest_author(self):
+        self.is_current_user_or_guest_author = True
+        if not self.env.user._is_public() and (not self.author_id or self.author_id != self.env.user.partner_id):
+            self.is_current_user_or_guest_author = False
+        if self.env.user._is_public() and (not self.author_guest_id or self.author_guest_id != self.env.context['guest']):
+            self.is_current_user_or_guest_author = False
```

Comment:
False by default, True only if correct

But ideally no default, make the code write only once the correct value directly.

Also it needs to be a recordset multi method